src/Document/CategoryManager.php

```diff
@@ -157,36 +106,85 @@ public function getPager(array $criteria, int $page, int $limit = 10, array $sor
         return $pager;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getRootCategoryWithChildren(CategoryInterface $category): void
+    public function getRootCategoryWithChildren(CategoryInterface $category)
     {
-        throw new \RuntimeException('Not Implemented yet');
+        if (null === $category->getContext()) {
+            throw new \InvalidArgumentException('Context cannot be null.');
+        }
+        if (null !== $category->getParent()) {
+            throw new \InvalidArgumentException('Method can be called only for root categories.');
+        }
+
+        $context = $category->getContext();
+
+        $this->loadCategories($context);
+
+        foreach ($this->categories[$context->getId()] as $contextRootCategory) {
+            if ($category->getId() === $contextRootCategory->getId()) {
+                return $contextRootCategory;
+            }
+        }
+
+        throw new \InvalidArgumentException(sprintf('Category "%s" does not exist.', $category->getId()));
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getRootCategoriesForContext(?ContextInterface $context = null): void
+    public function getRootCategoriesForContext(?ContextInterface $context = null)
     {
-        throw new \RuntimeException('Not Implemented yet');
+        $context = $this->getContext($context);
+
+        $this->loadCategories($context);
+
+        return $this->categories[$context->getId()];
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getAllRootCategories($loadChildren = true): void
+    public function getAllRootCategories($loadChildren = true)
     {
-        throw new \RuntimeException('Not Implemented yet');
+        $class = $this->getClass();
+
+        $rootCategories = $this->getObjectManager()->createQuery(sprintf('SELECT c FROM %s c WHERE c.parent IS NULL', $class))
```

Comment:
I guess this has never been used, maybe it's better to leave it with the exception?

It is supposed to be a `DocumentManager` and there is no `DocumentManager::createQuery` method.

Reply:
Didn't know, since there was already these call to `createQuery`.
https://github.com/sonata-project/SonataClassificationBundle/blob/master/src/Document/CategoryManager.php#L103
https://github.com/sonata-project/SonataClassificationBundle/blob/master/src/Document/CategoryManager.php#L201

What would be the right way to implement this ?

Reply:
Yeah, I meant that has never used because it is the same code it was before. 

I've never used this bundle, so I have no idea of its models and associations, so I'd rather throw an exception for now instead of adding some new untested code (that could be added in `3.x`).